src/JsParsing/PeastJsExpressionParser.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\VueJsTemplating\JsParsing;
+
+use Peast\Syntax\Node\CallExpression;
+use Peast\Syntax\Node\Expression;
+use Peast\Syntax\Node\Identifier;
+use Peast\Syntax\Node\MemberExpression;
+use Peast\Syntax\Node\ObjectExpression;
+use Peast\Syntax\Node\StringLiteral as PeastStringLiteral;
+use Peast\Syntax\Node\UnaryExpression;
+
+use RuntimeException;
+
+class PeastJsExpressionParser {
```

Comment:
To me this class looks confusing – in conjunction with its subclasses, yes, it’s a parser, but everything in this class itself is I wouldn’t call parsing: it takes an already parsed Peast expression and converts/translates/compiles/whatevers it to our own expression class.

I think the approach of keeping our own expression classes and converting the Peast AST to them makes sense (Peast also supports the visitor pattern via [tree traversing](https://github.com/mck89/peast/blob/master/doc/tree-traversing.md), but I get the impression that this isn’t a good fit for evaluating the AST), but perhaps we could move all this conversion code to a separate class?

Reply:
Fair. I've created a `PeastExpressionConverter` class to handle this now.